src/solvers/dgsem_tree/container_viscous_2d.jl

```diff
@@ -0,0 +1,66 @@
+mutable struct ViscousContainer2D{uEltype <: Real}
+    u_transformed::Array{uEltype, 4}
+    # Using an outer fixed-size datastructure leads to nasty implementations,
+    # avoid this here.
+    gradients::Vector{Array{uEltype, 4}}
+    flux_viscous::Vector{Array{uEltype, 4}}
+
+    # internal `resize!`able storage
+    _u_transformed::Vector{uEltype}
+    # Use Tuple for outer, fixed-size datastructure
+    _gradients::Tuple{Vector{uEltype}, Vector{uEltype}}
+    _flux_viscous::Tuple{Vector{uEltype}, Vector{uEltype}}
```

Comment:
Can you check if there is a noticeable performance difference between using `Tuple` instead of `Vector` for the outer container? If not, then I'd suggest sticking with `Vector` for consistency.

Reply:
Hm, for the Shear-Layer AMR case above the version using `Vector` is on average 3% slower than the one using `Tuple`. In this test, I called the AMR callback every step to increase the sample size.

I am not sure what conclusion we draw from this - on the one hand, we have only minor improvements - on the other hand these come essentially for free...

Reply:
Hm. 3% is not insignificant...

For now, let's use both mutable and immutable containers. Can you open an issue for this, specifying what lines of code prevent us from using Tuples everywhere?

Reply:
Yes, this seems to be the case.

Reply:
Yes, I will open an issue for this. It is definitely worth-while to check this for other problems.